### ISSUE TITLE
[SD3] pipe.enable_xformers_memory_efficient_attention

### ISSUE THREAD

**Issue by CanvaChen** [huggingface/diffusers#8535]
### Describe the bug

> RuntimeError: The size of tensor a (154) must match the size of tensor b (2304) at non-singleton dimension 1

### Reproduction

```python
# StableDiffusion3Pipeline
pipe.enable_xformers_memory_efficient_attention()
```

### System Info

diffusers==0.29.0
python=3.10
pytorch=2.3

### Who can help?

@yiyixuxu @DN6 @sayakpaul

**Comment by sayakpaul** [comment]
I don’t think we allow xformers attention in the SD3 blocks. Would you be interested in opening a PR? We will be happy to guide you.

**Comment by sayakpaul** [comment]
Oh that is okay. Here are a couple of reference pointers for you:

* This is the standard xformers attention processor class: https://github.com/huggingface/diffusers/blob/f96e4a16adb4c31bab4c0a3d0d145ed2b086ecb0/src/diffusers/models/attention_processor.py#L1312
* So, similarly, you would have to implement one for the joint attention processor block. More specifically, you will have to use ops from `xformers` in place of the native PyTorch ops here: https://github.com/huggingface/diffusers/blob/f96e4a16adb4c31bab4c0a3d0d145ed2b086ecb0/src/diffusers/models/attention_processor.py#L1135 (making sure the dimensions satisfy the criteria of xformers). 

Does this seem like a feature you would be interested in contributing? Not only we would greatly appreciate it but also help you by providing guidance #8276 

Let us know.

**Comment by sayakpaul** [comment]
That would be great! As mentioned we will be more than happy to guide you throughout.

**Comment by sayakpaul** [comment]
Thanks for your updates!

Do you want to open a PR with your implementation and tag myself and @yiyixuxu there?

It is perfectly okay to have it in an incomplete state.

**Comment by sayakpaul** [comment]
I think `None` mask param value is fine. 

Here is an example of how you can set the right processor:
https://github.com/huggingface/diffusers/blob/a899e42fc78fbd080452ce88d00dbf704d115280/src/diffusers/models/attention_processor.py#L381

**Comment by a-r-r-o-w** [comment]
@sayakpaul Do we still want this? There was another discussion opened today regarding this? https://github.com/huggingface/diffusers/discussions/9681

I think there were plans of removing xformers support within the library for the future, but it would still be compatible with `set_attn_processor()` methods for users who prefer to use it, I think.

**Comment by sayakpaul** [comment]
Sure. @DN6 could you give a final review of https://github.com/huggingface/diffusers/pull/8583?
